src/plugins/avoidance/include/plugins/avoidance/avoidance.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class AvoidanceImpl;
+class System;
+
+/**
+ * @brief The Avoidance class creates an interface for automatic-mode obstacle avoidance
+ */
+class Avoidance : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto avoidance = std::make_shared<Avoidance>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit Avoidance(System& system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~Avoidance();
+
+    /**
+     * @brief Possible results returned for avoidance requests.
+     */
+    enum class Result {
```

Comment:
I guess we will want an `UNKNOWN` here as well (at least other plugins usually do)

Reply:
Will do.